transport/nio/src/main/java/org/kaazing/gateway/transport/nio/internal/AbstractNioAcceptor.java

```diff
@@ -377,23 +377,19 @@ public UnbindFuture unbind(final ResourceAddress address) {
         // a hang when there's a race with Gateway.destroy().
         if (!unbindScheduler.isShutdown()) {
             final UnbindFuture future = new DefaultUnbindFuture();
-            unbindScheduler.submit(new FutureTask<>(new Runnable() {
-                @Override
-                public void run() {
-                    try {
-                        unbindInternal(address);
-                        future.setUnbound();
-                    } catch (ThreadDeath td) {
-                        throw td;
-                    } catch (Throwable t) {
-                        LOG.error("Exception during unbinding:\n"+address, t);
-                        //System.out.println("Exception during unbinding:\n"+address+"\n");
-                        //t.printStackTrace(System.out);
-                        future.setException(t);
-                        throw new RuntimeException(t);
-                    }
-                }
-            }, Boolean.TRUE));
+            try {
+                unbindInternal(address);
+                future.setUnbound();
+            } catch (ThreadDeath td) {
+                throw td;
+            } catch (Throwable t) {
+                LOG.error("Exception during unbinding:\n"+address, t);
+                //System.out.println("Exception during unbinding:\n"+address+"\n");
+                //t.printStackTrace(System.out);
+                future.setException(t);
+                throw new RuntimeException(t);
+            }
```

Comment:
I think that this change doesn't fix the issue. 
1. bindAsync releases bindLock and tries acquire it once the bind succeeded.
2. Between those two states, unbind acquires the bindLock and doesn't progress. 

unbind doesn't progress until bind thread NioServerBoss#process(Selector) which fires the channelUnregisteredFutures. But bind thread is blocked in 1, hence there is a deadlock.